tools/scripts/rebuildAnimationLibraryManifest.rb

```diff
@@ -33,50 +33,151 @@
-    metadata_progress_bar = ProgressBar.create(total: animations_by_name.size) unless @options[:verbose] || @options[:quiet]
-    animation_metadata_by_name = Hash[Parallel.map(animations_by_name, finish: lambda do |_, _, _|
-      metadata_progress_bar.increment unless @options[:verbose] || @options[:quiet]
-    end) do |name, objects|
-      # TODO: Validate that every JSON is paired with a PNG and vice-versa
+    Parallel.map(animation_objects.keys, finish: lambda do |name, _, result|
+      # This lambda runs synchronously after each entry is done processing - it's
+      # used to collect up results and warnings to the original process/thread.
+      if result.is_a? Hash
+        animation_metadata_by_name[name] = result
+      else
+        @warnings.push result
+      end
+      metadata_progress_bar.increment unless metadata_progress_bar.nil?
+    end) do |name|
+      # This is the parallel block.  This block should return a string to
+      # generate a warning and skip the animation, and a metadata Hash in
+      # the success case.
+      objects = animation_objects[name]
+
+      # Drop this animation if it is missing its JSON or PNG components
+      if objects['json'].nil?
+        next "Animation #{name} does not have a JSON file and was skipped."
+      elsif objects['png'].nil?
+        next "Animation #{name} does not have a PNG file and was skipped."
+      end
+
       # Actually download the JSON from S3
-      json_response = objects['json'].get
-      metadata = JSON.parse(json_response.body.read)
+      begin
+        json_response = objects['json'].get
+        metadata = JSON.parse(json_response.body.read)
+      rescue Aws::Errors::ServiceError => service_error
+        next <<-WARN
+There was an error retrieving #{name}.json from S3:
+#{service_error}
+The animation has been skipped.
+        WARN
+      rescue JSON::JSONError => json_error
+        next <<-WARN
+There was an error parsing #{name}.json:
+#{json_error}
+The animation ha been skipped.
+        WARN
+      end
+
+      # Default name to file name
+      next "Animation #{name} is missing the 'name' attribute." unless metadata['name'].is_a?(String)
 
       # If no frameCount information is present, it's probably a single frame
-      metadata['frameCount'] ||= 1
+      next "Animation #{name} is missing the 'frameCount' attribute." unless metadata['frameCount'].is_a?(Integer)
+
+      # Require frameSize information
+      next "Animation #{name} is missing the 'frameSize' attribute." unless metadata['frameSize'].is_a?(Hash)
+      next "Animation #{name} is missing the 'frameSize.x' attribute." unless metadata['frameSize']['x'].is_a?(Integer)
+      next "Animation #{name} is missing the 'frameSize.y' attribute." unless metadata['frameSize']['y'].is_a?(Integer)
+
+      # Require boolean looping attribute
+      next "Animation #{name} is missing the 'looping' attribute." unless [TrueClass, FalseClass].include? metadata['looping'].class
```

Comment:
Are we doing anything interesting here to have animations loop and single frames not loop? We probably don't need to, but if it's easy it might be a nice default.